spec/authorization_spec.rb

```diff
@@ -271,4 +271,38 @@ def to_params(*args, **kwargs, &block)
       expect(Controller.new(user, action, params).permitted_attributes(post, :revise).to_h).to eq("body" => "blah")
     end
   end
+
+  describe "#pundit_reset!" do
+    let(:new_user) { double }
+
+    it "clears the current user" do
+      expect(controller.pundit.user).to eq controller.current_user
+
+      controller.current_user = new_user
+      expect(controller.pundit.user).not_to eq controller.current_user
+
+      controller.pundit_reset!
+      expect(controller.pundit.user).to eq controller.current_user
+    end
```

Comment:
This is something I missed my last review. Apologies!

I think the spirit of this test is good, it captures the intent. I believe it's worth adding one more test an abstraction level above it, to capture the very important property that _authorize_ itself also resets.

However, `#user` is kinda unofficially public API so we should be sure to test that too.

```suggestion
    it "allows authorize to react to a user change" do
      expect(controller.authorize(post)).to be_truthy

      controller.current_user = double
      controller.pundit_reset!
      expect { controller.authorize(post) }.to raise_error(Pundit::NotAuthorizedError)
    end
    
    it "clears the pundit context user" do
      expect(controller.pundit.user).to be(user)
      
      new_user = double
      controller.current_user = new_user
      expect { controller.pundit_reset! }.to change { controller.pundit.user }.from(user).to(new_user)
    end
```

Finally, a code style nitpick. The `let(:new_user)` is only used in a single test here, but the `let` makes me believe it isn't. It's not necessary, it should be inlined in the test.